system/security/digisign/digisign.cpp

```diff
@@ -0,0 +1,224 @@
+            throw MakeStringException(-1, "digiSign:Cannot load private key file");
+
+#ifdef _USE_ZLIB
+        //create an RSA object from private key
+        BIO * keybio = BIO_new_mem_buf((void*) privateKeyBuff.str(), -1);
+        if (keybio == nullptr)
+        {
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiSign:BIO_new_mem_buf: %s", buff);
+        }
+
+        RSA * rsa = PEM_read_bio_RSAPrivateKey(keybio, nullptr, nullptr, nullptr);
+        if (rsa == nullptr)
+        {
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1,
+                    "digiSign:PEM_read_bio_RSAPrivateKey: %s", buff);
+        }
+        BIO_free_all(keybio);
+
+        //With the RSA object, create the digest and digital signature
+        EVP_MD_CTX * RSASignCtx = EVP_MD_CTX_create(); //allocate, initializes and return a digest context
+        EVP_PKEY * priKey = EVP_PKEY_new();
+        EVP_PKEY_assign_RSA(priKey, rsa);
+
+        //initialize context for SHA-256 hashing function
+        if (EVP_DigestSignInit(RSASignCtx, nullptr, EVP_sha256(), nullptr, priKey) <= 0)
+        {
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiSign:EVP_DigestSignInit: %s", buff);
+        }
+
+        //add string to the context
+        if (EVP_DigestSignUpdate(RSASignCtx, (size_t*)text, strlen(text)) <= 0)
+        {
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiSign:EVP_DigestSignUpdate: %s", buff);
+        }
+
+        //compute length of signature
+        size_t encMsgLen;
+        if (EVP_DigestSignFinal(RSASignCtx, nullptr, &encMsgLen) <= 0)
+        {
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiSign:EVP_DigestSignFinal1: %s", buff);
+        }
+
+        //compute signature (signed digest)
+        unsigned char * encMsg = (unsigned char*) malloc(encMsgLen);
+        if (EVP_DigestSignFinal(RSASignCtx, encMsg, &encMsgLen) <= 0)
+        {
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiSign:EVP_DigestSignFinal2: %s", buff);
+        }
+
+        //convert to base64
+        JBASE64_Encode(encMsg, encMsgLen, b64Signature, false);
+
+        //cleanup
+        free(encMsg);
+        EVP_PKEY_free(priKey);
+        EVP_MD_CTX_destroy(RSASignCtx);
+        return true;//success
+#else
+        throw MakeStringException(-1, "digiSign:Platform built without ZLIB");
+        return false;
+#endif
+    }
+
+
+    //Verify the given text was used to create the given digital signature
+    bool digiVerify(const char * text, const char * b64Signature)
```

Comment:
digiSign() returns a StringBuffer b64Signature, but digiVerify() takes a char *buffer, could these be more consistent ?